src/Forms/ChooseEncoding.cs

```diff
@@ -31,23 +36,22 @@ public void Initialize(string fileName)
         {
             try
             {
-                var file = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
-
-                int length = (int)file.Length;
-                if (length > 100000)
-                    length = 100000;
-
-                file.Position = 0;
-                _fileBuffer = new byte[length];
-                file.Read(_fileBuffer, 0, length);
-
-                for (int i = 0; i < length; i++)
+                using (var file = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
                 {
-                    if (_fileBuffer[i] == 0)
-                        _fileBuffer[i] = 32;
+                    int length = (int)file.Length;
+                    if (length > textBoxPreview.MaxLength)

```

Comment:
TextBox's default MaxLength is **32767** which means we don't need to buffer size above that